crates/sargon/src/system/drivers/networking_driver/support/test/mock_networking_driver.rs

```diff
@@ -25,10 +25,26 @@ impl MockNetworkingDriver {
         }
     }
 
+    pub fn with_spy_multiple_bodies(
+        status: u16,
+        bodies: Vec<BagOfBytes>,
+        spy: fn(NetworkRequest) -> (),
+    ) -> Self {
+        Self {
+            hard_coded_status: status,
+            hard_coded_bodies: Mutex::new(bodies),
+            spy,
+        }
+    }
+
     pub fn new(status: u16, body: impl Into<BagOfBytes>) -> Self {
         Self::with_spy(status, body, |_| {})
     }
 
+    pub fn new_with_bodies(status: u16, bodies: Vec<BagOfBytes>) -> Self {
+        Self::with_spy_multiple_bodies(status, bodies, |_| {})
+    }
```

Comment:
Seems you could use the `with_responses` I added instead, no?

Reply:
Not really. `with_responses` takes a Vec of a specific type, while I need different types of responses so I pass the serialized objects directly.